src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/interceptor/CustomUserAuthenticationInterceptor.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.interceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.lang.NonNull;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.util.security.AuthorisationUtil;
+import uk.gov.companieshouse.api.util.security.Permission;
+import uk.gov.companieshouse.api.util.security.SecurityConstants;
+import uk.gov.companieshouse.api.util.security.TokenPermissions;
+import uk.gov.companieshouse.limitedpartnershipsapi.utils.ApiLogger;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.ERIC_REQUEST_ID_KEY;
+
+public class CustomUserAuthenticationInterceptor implements HandlerInterceptor {
```

Comment:
I wouldn't have the 'Custom' in the class name, I'm thinking every class we write will be a custom one as we've written it for this service.

Reply:
What would you propose here I don't want to call it by the same name as the security library as that confused me with ROE. At first sight it looked like we had written it and forgotten about the one in the library, but it was a custom class required because of a problem with payment. I think that should be emphasised here.